02_TrackWeb/src/background.ts

```
@@ -0,0 +1,300 @@
+});
+
+// Listen for window focus changes
+chrome.windows.onFocusChanged.addListener((windowId) => {
+  if (windowId === chrome.windows.WINDOW_ID_NONE) {
+    // No window is focused, stop active timer
+    // Clear the active time alarm
+    chrome.alarms.clear(ACTIVE_TIME_ALARM_NAME);
+  } else if (activeTabId !== null) {
+    // A window is focused, check if the previously active tab is still active in a focused window
+     chrome.tabs.query({ active: true, windowId: windowId }, async (tabs) => {
+       if (tabs && tabs[0] && tabs[0].id === activeTabId && tabs[0].url && !isUrlIgnored(tabs[0].url)) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)